tests/pages/dashboard.py

```diff
@@ -0,0 +1,122 @@
+    _first_details_diff_locator = (By.CSS_SELECTOR,
+                                  '.dashboard-table tbody .dashboard-row:first-child + .dashboard-detail .diff')
+    _details_locator = (By.CSS_SELECTOR, '.dashboard-detail')
+
+    @property
+    def is_ip_toggle_present(self):
+        try:
+            ip_toggle = self.find_element(*self._ip_toggle_locator)
+            return True
+        except:
+            return False
+
+    @property
+    def first_row(self):
+        first_row = self.find_element(*self._first_row_locator)
+        return DashboardRow(self, root=first_row)
+
+    @property
+    def first_row_id(self):
+        return self.find_element(*self._first_row_locator).get_attribute('data-revision-id')
+
+    @property
+    def details_items_length(self):
+        details_items = self.find_elements(*self._details_locator)
+        return len(details_items)
+
+    def open_first_details(self):
+        first_row = self.find_element(*self._first_row_locator)
+        first_row.click()
+        self.wait.until(lambda s: len(self.find_elements(*self._details_locator)) > 0)
+
+    @property
+    def is_first_details_displayed(self):
+        first_details = self.find_element(*self._first_details_locator)
+        return first_details.is_displayed()
+
+    @property
+    def is_first_details_diff_displayed(self):
+        first_details_diff = self.find_element(*self._first_details_diff_locator)
+        return first_details_diff.is_displayed()
+
+    def click_page_two(self):
+        revsion_filter_form = self.find_element(*self._revision_filter_form_locator)
+        page_two_link = self.find_element(*self._page_two_link)
+        page_two_link.click()
+        # revsion-page updates to not 1
+        self.wait.until(lambda s: int(self.find_element(*self._revision_page_input).get_attribute('value')) is not 1)
+        # form is disabled when ajax request made
+        self.wait.until(lambda s: 'disabled' in revsion_filter_form.get_attribute('class'))
+        # wait for tray to be added
+        self.wait.until(lambda s: len(self.find_elements(*self._notification_tray_locator)) > 0)
+        # wait for notification in tray
+        self.wait.until(lambda s: len(self.find_elements(*self._first_notification_locator)) > 0)
+        # form editable when ajax response arrives
+        self.wait.until(lambda s: 'disabled' not in revsion_filter_form.get_attribute('class'))
+        # wait for notification to close
+        self.wait.until(lambda s: 'closed' in self.find_element(*self._first_notification_locator).get_attribute('class'))
+        # revsion-page-value updates to 1
+        self.wait.until(lambda s: int(self.find_element(*self._revision_page_input).get_attribute('value')) == 1)
+        # opacity maniulation finishes
+        self.wait.until(lambda s: 'opacity' not in self.find_element(*self._parent_locator).get_attribute('style'))
```

Comment:
Nice usage of `wait` statements to ensure the behavior of the page matches your expectations.